PlutoHelperAgent/ProjectLockerAndKeychainFunctions.m

```diff
@@ -89,156 +92,123 @@ + (NSArray *) load_data_from_keychain {
+                                                                       fromData:data completionHandler:^(NSData *data,NSURLResponse *response,NSError *error) {
+                                                                           NSString *datastring = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
+                                                                           NSLog(@"Data %@", datastring);
+                                                                           NSLog(@"Response %@", response);
+                                                                           if (error != NULL) {
+                                                                               NSLog(@"Error %@", error);
+                                                                           } else {
+                                                                               communicationStatus = 0;
+                                                                           }
+                                                                           
+                                                                       }];
+            
+            [uploadTask resume];
+        }
+        
+        sleep(1);
+        
+    } else {
+        
         NSURLSessionDataTask *uploadTask = [session dataTaskWithRequest:request completionHandler:^(NSData *data,NSURLResponse *response,NSError *error) {
             NSString *datastring = [[NSString alloc] initWithData:data encoding:NSUTF8StringEncoding];
             NSLog(@"Data %@", datastring);
             NSLog(@"Response %@", response);
             if (error != NULL) {
                 NSLog(@"Error %@", error);
-                
             }
             
-            NSString *compareThisPartOfTheString = [datastring substringToIndex:14];
-            
-            if ([compareThisPartOfTheString isEqual: @"{\"status\":\"ok\""]) {
-                
-                connectionStatus = 1;
+            if (test_connection == 1) {
+                NSString *compareThisPartOfTheString = [datastring substringToIndex:14];
                 
+                if ([compareThisPartOfTheString isEqual: @"{\"status\":\"ok\""]) {
+                    connectionStatus = 0;
+                }
             }
             
         }];
         
         [uploadTask resume];
         
+        sleep(1);
+        
     }
     
-    sleep(1);
-    
-    return connectionStatus;
+    return communicationStatus;
     
 }
 
-+ (NSString *) login_to_project_server {
++ (int) check_logged_in {
     
-    NSArray *dataFromKeychain = [self load_data_from_keychain];
+    connectionStatus = 1;
     
-    NSString *URLToUse = [NSString stringWithFormat: @"%@/api/login", [[NSUserDefaults standardUserDefaults] stringForKey:@"project_locker_url"]];
+    [self communicate_with_server:@"%@/api/isLoggedIn" :@"GET" :@"application/json" :NULL :1 :1];
```

Comment:
you could avoid using the literal %@ by simply having seperate parameters for server address and path?